src/sentry/ratelimits/sliding_windows.py

```diff
@@ -0,0 +1,209 @@
+"""
+The ratelimiter used by the metrics string indexer to rate-limit DB writes.
+
+As opposed to the API rate limiter, a limit in the "sliding window" rate
+limiter such as "10 requests / minute" does not reset to 0 every minute.
+Instead each window can be configured with a "granularity" setting such that
+the window gradually resets in steps of `granularity` seconds.
+
+Additionally this rate-limiter is not coupled to per-project/organization
+scopes, and can apply multiple sliding windows at once. On the flipside it is
+not strongly consistent and depending on usage it is very easy to over-spend
+quota, as checking quota and spending quota are two separate steps.
+"""
+
+from dataclasses import dataclass
+from time import time
+from typing import Any, Iterator, Optional, Sequence, Tuple
+
+from sentry.exceptions import InvalidConfiguration
+from sentry.utils import redis
+from sentry.utils.services import Service
+
+
+@dataclass(frozen=True)
+class Quota:
+    # The number of seconds to apply the limit to.
+    window: int
```

Comment:
I would suggest appending units to the variable names such that it's easier for the user. E.g`window_seconds`, `granularity_seconds`.

When looking at the tests I had to come back here to figure out what `window` and `granularity` were